### ISSUE TITLE
EMACS packages

### ISSUE THREAD

**Issue by eswenson1** [PDP-10/its#250]
There are lots of emacs packages in the PI distribution that are not in the DB build. The list is long:

- [x] ABSTR  :EJ
- [ ] ADA    :EJ
- [x] AUTO-S :EJ
- [x] AUX    :EJ
- [ ] BABYL  :EJ
- [ ] BABYL' :EJ
- [ ] BABYL- :EJ
- [ ] BABYLM :EJ
- [ ] BACKQ  :EJ
- [x] BARE   :EJ
- [ ] BIBLIO :EJ
- [ ] BIBTEX :EJ
- [ ] BLISS  :EJ
- [ ] CACHE  :EJ
- [x] CCL    :EJ (part of [PRFY])
- [ ] CLU    :EJ
- [ ] COLUMN :EJ
- [ ] COMPLT :EJ
- [x] DELIM  :EJ
- [x] DIRED  :EJ
- [ ] DM3025 :EJ
- [ ] DOCIX  :EJ
- [x] DOCLSP :EJ
- [x] DOCOND :EJ
- [ ] DOCTOR :EJ
- [ ] EENV   :EJ
- [x] EINIT  :EJ
- [ ] EMOUSE :EJ
- [ ] FIXLIB :EJ
- [ ] FORTRA :EJ
- [ ] GRIND  :EJ
- [ ] HAKLIB :EJ
- [ ] HEATH  :EJ
- [ ] HISTOR :EJ
- [ ] IBM370 :EJ
- [x] INFO   :EJ
- [x] IVORY  :EJ
- [ ] JLKMAC :EJ
- [ ] JOURNA :EJ
- [ ] JUSTIF :EJ
- [x] KBDMAC :EJ
- [ ] LABELS :EJ
- [ ] LEDEFS :EJ
- [x] LEDIT  :EJ
- [ ] LEDIT* :EJ
- [x] LISPT  :EJ
- [x] LSPUTL :EJ
- [ ] LUNAR  :EJ
- [ ] MAXIMA :EJ
- [x] MAZLIB :EJ
- [ ] MINI   :EJ
- [ ] MKDUMP :EJ
- [ ] MODE2  :EJ
- [x] MODLIN :EJ
- [ ] MQREPL :EJ
- [x] MUDDLE :EJ
- [ ] MWIND  :EJ
- [ ] NCASE  :EJ
- [ ] NCOLUM :EJ
- [ ] NEWS   :EJ
- [ ] NOVICE :EJ
- [ ] OUTLIN :EJ
- [x] PAGE   :EJ
- [ ] PASCAL :EJ
- [ ] PHRASE :EJ
- [x] PICTUR :EJ
- [x] PL1    :EJ
- [x] PURIFY :EJ
- [ ] QSEND  :EJ
- [ ] RENUM  :EJ
- [x] RMAIL  :EJ
- [ ] RUNOFF :EJ
- [ ] SAVMAC :EJ
- [ ] SCHEDI :EJ
- [ ] SCHEME :EJ
- [x] SCRIBE :EJ
- [x] SCRLIN :EJ
- [ ] SENDS  :EJ
- [ ] SKETCH :EJ
- [x] SLOWLY :EJ
- [x] SORT   :EJ
- [ ] SPLIT  :EJ
- [x] TAGGEN :EJ
- [x] TAGS   :EJ
- [x] TDEBUG :EJ
- [x] TEX    :EJ
- [x] TEXMAC :EJ
- [x] TIME   :EJ
- [x] TMACS  :EJ
- [ ] TPARSE :EJ
- [ ] TXJ    :EJ
- [x] VARG   :EJ
- [ ] VBABYL :EJ
- [ ] VMAIL  :EJ
- [x] VT100  :EJ
- [x] VT52   :EJ
- [ ] WEBSTE :EJ
- [x] WORDAB :EJ
- [ ] XHOST  :EJ
- [ ] XINDEN :EJ
- [x] XLISP  :EJ
- [ ] XLMS   :EJ
- [ ] XPAGE  :EJ
- [ ] Z      :EJ
- [ ] ZBABYL :EJ
- [ ] ZMENU  :EJ
- [ ] ZMOUSE :EJ

**Comment by larsbrinkhoff** [comment]
Yes.  I have done a few of them.  I'd like to pick the "most important" onces first.  Are there any you are missing in particular?

**Comment by eswenson1** [comment]
AUTO-S and DIRED are the only ones I'm really missing now.

**Comment by larsbrinkhoff** [comment]
Don't know where EENV :EJ comes from. Missing source?

**Comment by eswenson1** [comment]
Must be. Can't find it either.  From looking at EENV :EJ, it seems this package is to set up an emacs environment some special way with various packages loaded and settings. Not sure how important it is.  Have you found any documentation on it?

**Comment by larsbrinkhoff** [comment]
No, I just went through the list, looking for interesting libraries.  Didn't see any source code for EENV.  It should have a function called "Setup EENV", but there was no such string in any file anywhere, except the :EJ.

**Comment by eswenson1** [comment]
Yup. I looked for the same.

**Comment by larsbrinkhoff** [comment]
More missing source?

BIBTEX, INFEMX, JLKMAC, JOURNA, JUSTIF, LEDEFS, QSEND, XHOST, Z.

**Comment by larsbrinkhoff** [comment]
Keyboard macros are handy, no?  KBDMAC defines C-x  (, C-x ), C-x e.

**Comment by larsbrinkhoff** [comment]
These EMACS libraries was found on a TOPS-20 DUMPER tape:
ambassador
awindow
basic20
bufed
c108
capl
complain
givelib
grade
haz1510
heath
heath2
heath3
help-scan
inter
interlisp-mode
intermacs
kbdalt
list
lptlib
marker
merge
micro
micro-dat
mwindow
old-pcl
outline
outline-mode
pager
personal
picture
prbuf
prolog
ratfor
sbasic
schedit
send
send-mail
sftran3
smallsearch
stan
stanlib
teach-c100
tekheath
tekvt100
tvi9xx
underliner
vtxxx
wordcount
wplib
wrsome
xledit
xon-xoff

**Comment by eswenson1** [comment]
Source (and doc) as well as binary?

**Comment by larsbrinkhoff** [comment]
Yes.

**Comment by larsbrinkhoff** [comment]
@budd Has a VT2xx library: ftp://ftp.ultimate.com/pdp10/bucs20-anon/emacs/vt2xx.33

**Comment by larsbrinkhoff** [comment]
Here's a list of :EJ files in AI and MC, which are not in the EMACS directory:
AI: CPH; CPHLIB :EJ
AI: KMP; ZMENU :EJ
AI: KMP; TPARSE :EJ
AI: KMP; ZMOUSE :EJ
AI: DPH; EINIT :EJ
AI: MOON; LUNAR :EJ
AI: MAXTUL; MAXTAG :EJ
AI: JINX; JINXED :EJ
AI: PAO; LOCREG :EJ
AI: KLOTZ; QUICKR :EJ
AI: KLOTZ; EMACS :EJ
AI: EMACS1; SAIL :EJ
AI: IMSL; IMSMAC :EJ
AI: PDP11; PMACS :EJ
AI: DCB; DCBMAC :EJ
AI: LSB; FMODES :EJ
AI: CBF; CBFLIB :EJ
AI: AGRE; NEWWIN :EJ
AI: SPACY; HOSTAB :EJ
AI: SRA; ALICE :EJ
AI: RICH; UTILS :EJ
AI: DEVON; PDPTEN :EJ
AI: ALAN; EINIT :EJ
AI: GSB; LSTBUF :EJ
MC: MOON; LUNAR :EJ
MC: CBF; CBFLIB :EJ
MC: DCP; WINDO4 :EJ
MC: DCP; EINIT :EJ
MC: DCP; PHASE :EJ
MC: DCP; EENV :EJ
MC: SRA; ALICE :EJ
MC: ANBB; MSPLIT :EJ
MC: ANBB; POSTER :EJ

**Comment by eswenson1** [comment]
Turns out I had a copy of QSEND.  It is in EJS; QSEND 149

**Comment by rmaldersoniii** [comment]
You might want to look at the TOPS-20 library list on the system at LCM+L.

**Comment by bictorv** [comment]
When I started playing with the Github ITS (and not only the PI-based UP), I discovered that COMPLT is missing (source as EMACS1;COMPLT 5). TECO code being "binary", is there anything to consider before adding them to this repo?

Additionally, the standard incantation at the end of a personal `* EMACS` is 
```er emacs;emacs init◊ @y :m(hfx*)◊```
but the link from EMACS;EMACS INIT to EMACS;* EMACS is missing. A Github beginner might have asked how to add a link? ;-)

**Comment by larsbrinkhoff** [comment]
There shouldn't be anything special to consider.  Just add the source file to src, and build it like the others in emacs.tcl, and the timestamp in build/timestamps.txt.

Links can be stored version controlled in the source tree, or made by the build script using DDT's `:link` command.
